### PR TITLE
Add Contributor guide

### PR DESCRIPTION
**Pull request by weenxin**
## Purpose
Recently I have read the KubeDB/operator source code, I write a document to explain how KubeDB operator is implemented. Maybe some other need it.

Maybe there are some errors, Maybe It is totally useless, It is up to you to decide how to deal with this PR.

### CODE REVIEW

**Comment by hossainemruz** [comment]
Thank you @weenxin. We really appreciate it. :+1:

File: docs/contributor-guide/README.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# KubeDB Operator Design Document
```
Comment (the-redback) [inline docs/contributor-guide/README.md:1]: I would prefer some `Installation` and `testing` guide first before jumping into code itself.

**Review by the-redback — CHANGES_REQUESTED** [review]
The `images` folder should be moved to either `docs` or `docs/contribution-guide` folder.

**Comment by tamalsaha** [comment]
I have recorded a video showing we develop and test operators. https://youtu.be/vbOXlWjRjHM 

I am going to document more details as time permits. Feel free to ask me anything.

**Pull request closed without merging**